aws-android-sdk-ddb-mapper-test/src/androidTest/java/com/amazonaws/mobileconnectors/dynamodbv2/dynamodbmapper/BatchLoadIntegrationTest.java

```diff
@@ -217,6 +218,41 @@ public void testExponentialBackOffForBatchGetInMapper()
         }
     }
 
+    @Test
+    public void testGzippedResponse() throws InterruptedException {
+        ClientConfiguration clientConfiguration = new ClientConfiguration()
+                .withEnableGzip(true);
+        AmazonDynamoDBClient gzippingDynamo = new AmazonDynamoDBClient(credentials, clientConfiguration);
+        DynamoDBMapper gzippingMapper = new DynamoDBMapper(gzippingDynamo);
+
+        // To see whether batchGet can handle more than 100 items per request
+        final int numItems = 200;
+        List<NumberSetAttributeClass> objs = new ArrayList<NumberSetAttributeClass>();
+        List<KeyPair> keyPairs = new LinkedList<KeyPair>();
+        Class<?> clazz = null;
+        for (int i = 0; i < numItems; i++) {
+            NumberSetAttributeClass obj = getUniqueNumericObject();
+            objs.add(obj);
+            clazz = obj.getClass();
+            keyPairs.add(new KeyPair().withHashKey(obj.getKey()));
+        }
+
+        gzippingMapper.batchSave(objs);
+
+        Map<Class<?>, List<KeyPair>> itemsToGet = new HashMap<Class<?>, List<KeyPair>>();
+        Map<String, List<Object>> response = null;
+        itemsToGet.put(clazz, keyPairs);
+        response = gzippingMapper.batchLoad(itemsToGet);
+        List<Object> items = response.get(TABLE_NAME);
+        assertEquals(numItems, items.size());
+
+        for (Object item : items) {
+            objs.contains(item);
+        }
+        Thread.sleep(1000 * 10);
```

Comment:
Is this sleep necessary?

Reply:
I'm not actually sure--I cargo-cult copy/pasted it from the previous test :)

I'm inclined to leave it in for now, since it's reliable in the pipelines (which is why I assume it was added to begin with).